shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java

```diff
@@ -142,4 +162,27 @@ private void processingFailed(Class<?> entity, Throwable t) {
     private void processingComplete(Class<?> entity) {
         LOGGER.debug("[shard] Successfully processed all entities '{}' to deploy or delete", entity.getSimpleName());
     }
+
+    private Uni<HttpResponse<Buffer>> getAuthenticatedRequest(HttpRequest<Buffer> request, Function<HttpRequest<Buffer>, Uni<HttpResponse<Buffer>>> executor) {
+        if (client.isPresent()) {
```

Comment:
if sso id is active...

Reply:
@fjtirado I think we should always assume that the `OidcClient` or `Tokens` injection will be present. I don't think we should make it an Optional dependency.

Instead, the tests should provide the correct implementation for the `OidcClient` or `Tokens` e.g. it can be mocked directly via `Mockito`, or the better option is to use WireMock to stub the invocation to Keycloak and override the configuration as shown here: https://quarkus.io/guides/security-openid-connect-client#integration-testing-oidc-client

This means that we only ever have one path in the code, but we simply change the configuration of the `OidcClient` / `Tokens` that is injected, depending on the environment in which the code is executing.